src/client.ts

```diff
@@ -254,8 +155,7 @@ export function createClient<T extends ClientParams>(rawConfig: T): Client<T> {
     auth: ('authenticate' in httpWithAuth && httpWithAuth.authenticate
       ? {
           ...authService(httpWithAuth),
-          authenticate: (oauth: AuthParams) =>
-            httpWithAuth.authenticate(parseAuthParams(oauth)),
```

Comment:
This was basically the only thing I wanted to change, splitting up the `parseAuthParams` 😬